### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by RoBorg**

### CODE REVIEW

**Review by wongmrdev — COMMENTED** [review]
Do we need a full ubuntu here or can we get away with a node:18 image?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Copy the dalai source code into the container
+WORKDIR /dalai
+
+COPY . .
+
+# Install dalai and its dependencies
+RUN npm install
+
+# Install a model
+RUN npx dalai alpaca install 7B
```
Comment (ImLunaHey) [inline Dockerfile:29]: This should be using either build time or runtime vars so the user can pick the models they want.

**Comment by ImLunaHey** [comment]
This was also already done in https://github.com/cocktailpeanut/dalai/pull/39

**Comment by ImLunaHey** [comment]
@RoBorg from what I've seen those changes should be done since the paths need to be updated for the container.

**Comment by RoBorg** [comment]
@ImLunaHey I don't think so - what I have here works fine, and if you wanted to mount files from outside the container you could just mount a directory over the top by using something like `-v /my/path:/dalai/models`. The path inside the container shouldn't change.

At the end of the day, I think it would be great to get a Dockerfile merged in - whether it's this one or #39, I don't mind.
I think they take slightly different approches - this one ends up with the model being inside the container, #39 always mounts it in. I can see arguments for both cases... it's just up to a maintainer to pick which approach the prefer.

**Comment by metrafonic** [comment]
My implementation with a dockerfile and a docker compose: https://gist.github.com/metrafonic/8402d96a1348208d5857a357e2b77ad7
Bind mounts for models and so on

**Comment by metrafonic** [comment]
This branch should not be merged as is currently

**Comment by RoBorg** [comment]
See #229

**Pull request closed without merging**